collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -700,6 +708,44 @@ private void nonblockingCreateSavePointData() {
         }
     }
 
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
```

Comment:
This two methods still don't work like expected please take a look at onDateChanged/onRankingChanged
and also you probably don't need this method at all because there is existing onDateChanged can you use it?

Reply:
@grzesiek2010 I checked the code of `DatePickerDialog`. 
`    @Override
    public void onClick(@NonNull DialogInterface dialog, int which) {
        switch (which) {
            case BUTTON_POSITIVE:
                if (mDateSetListener != null) {
                    // Clearing focus forces the dialog to commit any pending
                    // changes, e.g. typed text in a NumberPicker.
                    mDatePicker.clearFocus();
                    mDateSetListener.onDateSet(mDatePicker, mDatePicker.getYear(),
                            mDatePicker.getMonth(), mDatePicker.getDayOfMonth());
                }
                break;
            case BUTTON_NEGATIVE:
                cancel();
                break;
        }
    }
`
`onDateSet` is called on clicking the positive button. Do I need to override the `onClick()` method?

Reply:
I think I don't understand...
In FixedDatePickerDialog you have:
```
private DatePickerDialog.OnDateSetListener listener;
```
but in CustomDatePickerDialog you have 
```
private CustomDatePickerDialogListener listener;
```
end then you end up with two different methods in FormEntryActivity:
onDateChanged and onDateSet I think you can share use just one listener probably:
```
private DatePickerDialog.OnDateSetListener listener;
```
because you won't be able to use `CustomDatePickerDialogListener` in `FixedDatePickerDialog`
is that clear?
This was just the first problem, another one is that your methods should work like onDateChanged/onRankingChanged
I mean they should use setData() otherwise you just pass an answer to the first question what won't work in case of having multiple questions on the same screen.
The difference between your approach and the one we need to have is that as I said you add an answer to the first question but you should do the same we do in `CustomDatePickerDialog` where we set `formController.setIndexWaitingForData()` and then using if you use setData the method is able to find the widget that is waiting for answer.

Reply:
Oh now I get it. Sorry for misunderstanding

Reply:
Great this looks good now.